<?xml version="1.0"?>
<launch>

  <arg name="manager"    		value="synkar_manager"/>
  <arg name="left"              value="camera/left"/>
  <arg name="right"             value="camera/right"/>
  <arg name="approximate_sync"  value="true" />
  <arg name="respawn"  			value="false" />

  <!-- Start a nodelet manager; the name must be unqualified here -->
  <node pkg="nodelet" type="nodelet" name= "$(arg manager)" args="manager" />

  <arg     if="$(arg respawn)" name="bond" value="" />
  <arg unless="$(arg respawn)" name="bond" value="--no-bond" />


    <node pkg="nodelet" type="nodelet" name="cv_camera_nodelet_left" args="load cv_camera/CvCameraNodelet $(arg manager)" output="screen">
        <param name="device_path" value="/dev/video1" />
        <param name="frame_id" value="$(arg left)" />
        <param name="rate" value="72.0" />
        <param name="image_width" value="1280" />
        <param name="image_height" value="720" />
        <param name="camera_info_url" value="file://$(find cv_camera)/resources/left.yaml" />
        <remap from="~image_raw" to="$(arg left)/image_raw"/>
        <remap from="~camera_info" to="$(arg left)/camera_info"/>
        <remap from="~set_camera_info" to="$(arg left)/set_camera_info"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="cv_camera_nodelet_right" args="load cv_camera/CvCameraNodelet $(arg manager)" output="screen">
        <param name="device_path" value="/dev/video0" />
        <param name="frame_id" value="$(arg right)" />
        <param name="rate" value="72.0" />
        <param name="image_width" value="1280" />
        <param name="image_height" value="720" />
        <param name="camera_info_url" value="file://$(find cv_camera)/resources/right.yaml" />
        <remap from="~image_raw" to="$(arg right)/image_raw"/>
        <remap from="~camera_info" to="$(arg right)/camera_info"/>
        <remap from="~set_camera_info" to="$(arg right)/set_camera_info"/>
    </node>
    
     <node pkg="nodelet" type="nodelet" name="rectify_color_left" args="load image_proc/rectify $(arg manager) $(arg bond)" respawn="$(arg respawn)">
    <remap from="image_mono" to="$(arg left)/image_raw" />
    <remap from="image_rect" to="$(arg left)/image_rect" />
    <remap from="camera_info" to="$(arg left)/camera_info" />
  </node>  

 <node pkg="nodelet" type="nodelet" name="rectify_color_right" args="load image_proc/rectify $(arg manager) $(arg bond)" respawn="$(arg respawn)">
    <remap from="image_mono" to="$(arg right)/image_raw" />
    <remap from="image_rect" to="$(arg right)/image_rect" />
    <remap from="camera_info" to="$(arg right)/camera_info" />
  </node> 
  
  
    <!--<node pkg="rosbag" type="record" name="bag_teste" args="/camera/left/image_raw /camera/right/image_raw -o /pendrive/teste_indoor.bag"/>-->
    

</launch>
